### ISSUE TITLE
final test_ initial

### ISSUE THREAD

**Issue by Annashuvarina24** [Annashuvarina24/Evaluation-metrics#13]
def final_test(model, df, target_column, strategy="gap", gap_hours=720) :
    """
    Train model on full training data and evaluate on test data with hourly splits.

    Parameters:
        model: The trained model.
        df: DataFrame containing full dataset.
        target_column: The column to forecast.
        strategy (str): Type of test set split ("gap", "sign_preserving").
        gap_hours (int): Hours to add as a gap if using "gap" strategy.
    """
    try :
        logger.info(f"Running final test for {target_column}...")

        # ✅ Define train and test split with an optional gap
        train_end_date = pd.to_datetime(CONFIG[ 'train_end' ])
        test_start_date = train_end_date + pd.Timedelta(hours=gap_hours)

        train_data = df.loc[ CONFIG[ 'train_start' ] :CONFIG[ 'train_end' ] ]
        test_data = df.loc[ test_start_date :CONFIG[ 'test_end' ] ]

        X_train, y_train = train_data.drop(columns=[ target_column ]), train_data[ target_column ]
        X_test, y_test = test_data.drop(columns=[ target_column ]), test_data[ target_column ]

        # ✅ Apply Scaling
        scaler_X = StandardScaler()
        X_train_scaled = scaler_X.fit_transform(X_train)
        X_test_scaled = scaler_X.transform(X_test)

        scaler_y = StandardScaler()
        y_train_scaled = scaler_y.fit_transform(y_train.values.reshape(-1, 1))
        y_test_scaled = scaler_y.transform(y_test.values.reshape(-1, 1))

        # ✅ Handle sign-preserving forecast separately
        if strategy == "sign_preserving" :
            X_train_pos, y_train_pos = X_train_scaled[ y_train > 0 ], y_train_scaled[ y_train > 0 ]
            X_train_neg, y_train_neg = X_train_scaled[ y_train < 0 ], y_train_scaled[ y_train < 0 ]

            X_test_pos, y_test_pos = X_test_scaled[ y_test > 0 ], y_test_scaled[ y_test > 0 ]
            X_test_neg, y_test_neg = X_test_scaled[ y_test < 0 ], y_test_scaled[ y_test < 0 ]

            model.fit(X_train_pos, y_train_pos)
            y_pred_pos = scaler_y.inverse_transform(model.predict(X_test_pos)).flatten()

            model.fit(X_train_neg, y_train_neg)
            y_pred_neg = scaler_y.inverse_transform(model.predict(X_test_neg)).flatten()

            y_pred = np.concatenate([ y_pred_pos, y_pred_neg ])
        else :
            model.fit(X_train_scaled, y_train_scaled)
            y_pred = scaler_y.inverse_transform(model.predict(X_test_scaled)).flatten()

        return evaluate_metrics(y_test, y_pred)

    except Exception as e :
        logger.error(f"Final test failed: {e}")
        raise
